### PR TITLE
PSAR Indicator

### PR DESCRIPTION
**Pull request by StaRenn**

### CODE REVIEW

File: src/psar.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        if (!this.result) {
+            this.result = close;
+            this.low1 = low;
+            this.high1 = high;
+            this.low2 = low;
+            this.high2 = high;
+            this.highest = high;
+            this.lowest = low;
+            this.accelerationForce = this.start;
```
Comment (StaRenn) [inline src/psar.ts]: There must be Factor, not Force, i will fix it tomorrow

File: src/psar.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    constructor(start = 0.02, acceleration = 0.02, max = 0.2) {
+        this.start = start;
+        this.acceleration = acceleration;
+        this.max = max;
+        this.result = 0;
+        this.isBullTrend = true;
+    }
+
+    nextValue({ h: high, l: low, c: close }: { h: number; l: number; c: number }) {
```
Comment (BusinessDuck) [inline src/psar.ts]: Don't use destructure assignment, a little bit slowly than pass separate arguments.

File: src/psar.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            if (high > this.psar) {
+                this.isBullTrend = true;
+                isReverse = true;
+                this.psar = this.lowest;
+                this.highest = high;
+                this.accelerationForce = this.start;
+            }
+        }
+
+        if (!isReverse) {
```
Comment (BusinessDuck) [inline src/psar.ts]: why reverse condition? Just swap if/else

File: src/psar.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                if (this.low1 < this.psar) {
+                    this.psar = this.low1;
+                }
+                if (this.low2 < this.psar) {
+                    this.psar = this.low2;
+                }
+            } else {
+                if (low < this.lowest) {
+                    this.lowest = low;
+                    this.accelerationForce = Math.min(this.accelerationForce + this.acceleration, this.max);
```
Comment (BusinessDuck) [inline src/psar.ts]: "Math.min" very slow, use if conditions like 
min = a > b ? b : a;

**Review by BusinessDuck — APPROVED** [review]
Looks very good, apply some fixes for comments and i'll merge PSAR to collection

**Comment by StaRenn** [comment]
fixed comments + added moment value test

**Pull request merged**